docs/03-extensions/paypal/index.mdx

```diff
@@ -0,0 +1,166 @@
+  extensions: [
+    themeChocolatine(),
+    magento2({ storesConfig }),
+    // highlight-next-line
+    paypal("magento2"), // ⚠️ needs to be after themeChocolatine()
+  ],
+  stores: storesConfig,
+  cache: cacheConfig,
+  configuration: {
+    providers: [],
+  },
+});
+```
+
+### Environment variables
+
+This extension requires environment variable to work properly. Please refer to
+the
+[Environment variables reference](/docs/3.x/extensions/paypal/reference/environment-variables)
+for more information.
+
+### Register your PayPal payment components
+
+:::note
+
+The current method for configuring payment components is an interim solution. We
+recognize that it may be cumbersome for users. Please be assured that this
+process will undergo significant improvements in the future, as Front-Commerce
+evolves to include more versatile extension points for Payment systems.
+
+:::
+
+1. Override the file that lets you register additional payments methods in
+   Front-Commerce
+
+   ```shell
+   mkdir -p app/theme/modules/Checkout/Payment/AdditionalPaymentInformation/
+   cp -u node_modules/@front-commerce/theme-chocolatine/theme/modules/Checkout/Payment/AdditionalPaymentInformation/getAdditionalDataComponent.js app/theme/modules/Checkout/Payment/AdditionalPaymentInformation/getAdditionalDataComponent.js
+   ```
+
+2. Register PaypalButton
+
+   ```diff title="app/theme/modules/Checkout/Payment/AdditionalPaymentInformation/getAdditionalDataComponent.js"
+   +import PaypalButton from "theme/modules/Checkout/Payment/AdditionalPaymentInformation/PaypalButton";
+
+   const ComponentMap = {
+   +  paypal_button: PaypalButton,
+   };
+
+   ...
+   ```
+
+After restarting Front-Commerce, you should be able to see a new payment method
+called "credit card" in your checkout step.
+
+### Advanced: customize data sent to Paypal
+
+:::caution WIP
+
+<span>
+  This advanced pattern must be documented with further details. While we are
+  working on it, please <ContactLink /> if you need further assistance.
+</span>
+
+:::
+
+The Paypal payment module is extensible. It leverages Front-Commerce's "data
+transform" pattern to allow developers to customize payloads sent to Paypal for
+Payer data and Payer units.
+
+Both the payer data and payer units objects can be customized at application
+level. It allows to add additional metadata depending on your own logic. For
+this, you can use the `registerPayerDataTransform` and
+`registerPurchaseUnitsDataTransform` methods of the Paypal loader to add your
+custom transformers.
+
+See the tests for an example (while a detailed documentation is being written):
+
+- https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/c1ca1ef8a60ecb545e2758d04a4d11577e764658/src/server/modules/payment-paypal/__pacts__/loaders.js#L230
+- https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/c1ca1ef8a60ecb545e2758d04a4d11577e764658/src/server/modules/payment-paypal/__pacts__/loaders.js#L281
+
```

Comment:
Update links to more recent commits like https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/fc-1660-paypal/packages/paypal/modules/common/__pacts__/loaders.js#L248

Reply:
I'll wait for the MR to be merged to avoid back and forth on this subject, but yes, good point.